Client/FeatureFlags/FeatureFlagsManager.swift

```diff
@@ -0,0 +1,78 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import Shared
+
+protocol FeatureFlagsProtocol { }
+
+extension FeatureFlagsProtocol {
+    var featureFlags: FeatureFlagsManager {
+        return FeatureFlagsManager.shared
+    }
+}
+
+/// An enum describing the featureID of all features, historical, avalable, or in development.
+enum FeatureFlagID: String, CaseIterable {
+    case chronologicalTabs
+    case inactiveTabs
+    case nimbus
+    case recentlySaved
+    case shakeToRestore
+}
+
+class FeatureFlagsManager {
+
+    /// This Singleton should only be accessed directly in places where the
+    /// `FeatureFlagsProtocol` is not available. Otherwise, access to the feature
+    /// flags system should be done through the protocol, giving access to the
+    /// `featureFlags` variable.
+    static let shared = FeatureFlagsManager()
+
+    private var profile: Profile!
+    private var features: [FlaggableFeature] = []
+
+    init() { }
+
+    public func isFeatureActive(_ featureID: FeatureFlagID) -> Bool {
+        for feature in features {
+            if feature.featureID == featureID {
+                return feature.isActive
+            }
+        }
```

Comment:
A thought, instead of doing a for loop on an array what do you think of having a dictionary with [FeatureID: Status]?

Reply:
Interesting. This is a great idea. I'll do that too. But slightly differently, maybe, for ease of reuse.